sympy/printing/glsl.py

```diff
@@ -0,0 +1,469 @@
+from sympy import Basic, Function, Symbol
+from sympy.printing.codeprinter import CodePrinter
+from sympy.core.function import _coeff_isneg
+from sympy.printing.precedence import precedence
+from sympy.core.compatibility import string_types, range
+from sympy.core import S
+from sympy.codegen.ast import Assignment
+from functools import reduce
+
+known_functions = {
+    'Abs': 'abs',
+    'sin': 'sin',
+    'cos': 'cos',
+    'tan': 'tan',
+    'acos': 'acos',
+    'asin': 'asin',
+    'atan': 'atan',
+    'atan2': 'atan',
+    'ceiling': 'ceil',
+    'floor': 'floor',
+    'sign': 'sign',
+    'exp': 'exp',
+    'log': 'log',
+    'add': 'add',
+    'sub': 'sub',
+    'mul': 'mul',
+    'pow': 'pow'
+}
+
+class GLSLPrinter(CodePrinter):
+    """
+    Rudimentary, generic GLSL printing tools.
+
+    Additional settings:
+    'use_operators': Boolean (should the printer use operators for +,-,*, or functions?)
+    """
+    _not_supported = set()
+    printmethod = "_glsl"
+    language = "GLSL"
+
+    _default_settings = {
+        'use_operators': True,
+        'mat_nested': False,
+        'mat_separator': ',\n',
+        'mat_transpose': False,
+        'glsl_types': True,
+
+        'order': None,
+        'full_prec': 'auto',
+        'precision': 32,
```

Comment:
`'precision'` doesn't actually refer to the number of bits used, but rather number of significant digits. Using 9 would be the right number for float32. We are working on better representation of types in the printers.